automation/Zhp.Office.AccountManagement/Zhp.Office.AccountManagement.Domain/Services/OldAccountCleaner.cs

```diff
@@ -0,0 +1,49 @@
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Zhp.Office.AccountManagement.Domain.Ports;
+
+namespace Zhp.Office.AccountManagement.Domain.Services
+{
+    public class OldAccountCleaner
+    {
+        private readonly ILastUsageRepository lastUsageRepo;
+        private readonly IAccountManager manager;
+        private readonly ILogger<OldAccountCleaner> logger;
+
+        public OldAccountCleaner(ILastUsageRepository oldUserFinder, IAccountManager manager, ILogger<OldAccountCleaner> logger)
+        {
+            this.lastUsageRepo = oldUserFinder;
+            this.manager = manager;
+            this.logger = logger;
+        }
+
+        public async Task CleanOldAccounts(uint thresholdInDays, CancellationToken token)
+        {
+            var utcDateThreshold = DateTime.UtcNow.Date.AddDays(-thresholdInDays);
+            var oldAccounts = await lastUsageRepo.FindLastActivity(token)
+                .Where(r => r.LastUsage == null || r.LastUsage < utcDateThreshold)
```

Comment:
Wydaje mi się, że będzie usuwać, bo wydaje mi się, że jeśli użytkownik się jeszcze nie zalogował na swoje konto - bo np. miał je stworzone w poprzednim przebiegu - to chyba będzie tu zwracać null właśnie?
Jest może opcja tutaj sprawdzenia daty założenia konta, żeby określić wiek minimalny kont do usunięcia tu?

Reply:
Dobre spostrzeżenie

Też mi to potem przeszło przez głowę i zapisałem sobie todo w pr. Mamy tam dostęp do daty przyznania licencji (vide testowy csv), myślę, że nie usuniemy kont które dostały licencję w ciągu 60 dni

Reply:
Wrzuciłem poprawkę